src/library/searchqueryparser.cpp

```diff
@@ -114,51 +115,46 @@ void SearchQueryParser::parseTokens(QStringList tokens,
             continue;
         }
 
+        bool negate = token.startsWith(kNegatePrefix);
+        std::unique_ptr<QueryNode> pNode;
+
         if (m_fuzzyMatcher.indexIn(token) != -1) {
             // TODO(XXX): implement this feature.
         } else if (m_textFilterMatcher.indexIn(token) != -1) {
-            bool negate = token.startsWith(kNegatePrefix);
             QString field = m_textFilterMatcher.cap(1);
             QString argument = getTextArgument(
                 m_textFilterMatcher.cap(2), &tokens).trimmed();
 
             if (!argument.isEmpty()) {
-                std::unique_ptr<QueryNode> pNode(std::make_unique<TextFilterNode>(
-                    m_database, m_fieldToSqlColumns[field], argument));
-                if (negate) {
-                    pNode = std::make_unique<NotNode>(std::move(pNode));
+                if (field == "crate") {
+                    pNode = std::make_unique<CrateFilterNode>(
+                          argument, &m_pTrackCollection->crates());
```

Comment:
I also noticed this, but didn't want to confuse you with minor issues ;) Method parameters should be ordered starting from broad, context-related  arguments to more specialized arguments. Ok, since we both agree let's fix it now.